error[E0107]: this struct takes 3 generic arguments but 2 generic arguments were supplied

Compiling libc v0.2.112
   Compiling proc-macro2 v1.0.33
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.82
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.0.1
   Compiling serde_derive v1.0.131
   Compiling serde v1.0.131
   Compiling cc v1.0.72
   Compiling log v0.4.14
   Compiling lazy_static v1.4.0
   Compiling version_check v0.9.3
   Compiling memchr v2.4.1
   Compiling futures-core v0.3.18
   Compiling slab v0.4.5
   Compiling futures-sink v0.3.18
   Compiling cfg-if v0.1.10
   Compiling bytes v0.5.6
   Compiling futures-channel v0.3.18
   Compiling scopeguard v1.1.0
   Compiling futures-task v0.3.18
   Compiling pin-project-lite v0.2.7
   Compiling pin-project-lite v0.1.12
   Compiling futures-util v0.3.18
   Compiling once_cell v1.8.0
   Compiling smallvec v1.7.0
   Compiling futures-io v0.3.18
   Compiling typenum v1.14.0
   Compiling pin-utils v0.1.0
   Compiling parking_lot_core v0.8.5
   Compiling getrandom v0.1.16
   Compiling tinyvec_macros v0.1.0
   Compiling pin-project-internal v0.4.28
   Compiling spin v0.5.2
   Compiling untrusted v0.7.1
   Compiling percent-encoding v2.1.0
   Compiling matches v0.1.9
   Compiling bitflags v1.3.2
   Compiling proc-macro-hack v0.5.19
   Compiling convert_case v0.4.0
   Compiling linked-hash-map v0.5.4
   Compiling unicode-bidi v0.3.7
   Compiling ppv-lite86 v0.2.15
   Compiling pkg-config v0.3.24
   Compiling crossbeam-utils v0.8.5
   Compiling unicode-segmentation v1.8.0
   Compiling base64 v0.12.3
   Compiling fnv v1.0.7
   Compiling itoa v0.4.8
   Compiling async-trait v0.1.52
   Compiling const_fn v0.4.8
   Compiling hashbrown v0.11.2
   Compiling bytes v1.1.0
   Compiling either v1.6.1
   Compiling match_cfg v0.1.0
   Compiling copyless v0.1.5
   Compiling ryu v1.0.9
   Compiling serde_json v1.0.73
   Compiling cpufeatures v0.2.1
   Compiling crc32fast v1.3.0
   Compiling httparse v1.5.1
   Compiling regex-syntax v0.6.25
   Compiling opaque-debug v0.3.0
   Compiling quick-error v1.2.3
   Compiling crossbeam-epoch v0.9.5
   Compiling encoding_rs v0.8.30
   Compiling rayon-core v1.9.1
   Compiling openssl v0.10.38
   Compiling adler v1.0.2
   Compiling itoa v1.0.1
   Compiling foreign-types-shared v0.1.1
   Compiling byteorder v1.4.3
   Compiling native-tls v0.2.8
   Compiling humantime v2.1.0
   Compiling base64 v0.13.0
   Compiling anyhow v1.0.51
   Compiling mime v0.3.16
   Compiling openssl-probe v0.1.4
   Compiling subtle v2.4.1
   Compiling language-tags v0.2.2
   Compiling traitobject v0.1.0
   Compiling utf-8 v0.7.6
   Compiling termcolor v1.1.2
   Compiling log-mdc v0.1.0
   Compiling arc-swap v0.4.8
   Compiling hex v0.4.3
   Compiling instant v0.1.12
   Compiling openssl-src v111.16.0+1.1.1l
   Compiling num-traits v0.2.14
   Compiling indexmap v1.7.0
   Compiling memoffset v0.6.5
   Compiling miniz_oxide v0.4.4
   Compiling num-integer v0.1.44
   Compiling rayon v1.5.1
   Compiling tracing-core v0.1.21
   Compiling lock_api v0.4.5
   Compiling input_buffer v0.3.1
   Compiling generic-array v0.14.4
   Compiling standback v0.2.17
   Compiling time v0.2.27
   Compiling cookie v0.14.4
   Compiling ring v0.16.20
   Compiling tinyvec v1.5.1
   Compiling brotli-sys v0.3.2
   Compiling form_urlencoded v1.0.1
   Compiling lru-cache v0.1.2
   Compiling yaml-rust v0.4.5
   Compiling heck v0.3.3
   Compiling http v0.2.5
   Compiling foreign-types v0.3.2
   Compiling fxhash v0.2.1
   Compiling unsafe-any v0.4.2
   Compiling openssl-sys v0.9.72
   Compiling unicode-normalization v0.1.19
   Compiling typemap v0.3.3
   Compiling net2 v0.2.37
   Compiling iovec v0.1.4
   Compiling signal-hook-registry v1.4.0
   Compiling num_cpus v1.13.0
   Compiling hostname v0.3.1
   Compiling time v0.1.44
   Compiling atty v0.2.14
   Compiling getrandom v0.2.3
   Compiling thread-id v3.3.0
   Compiling quote v1.0.10
   Compiling aho-corasick v0.7.18
   Compiling crossbeam-channel v0.5.1
   Compiling idna v0.2.3
   Compiling threadpool v1.8.1
   Compiling rand_core v0.5.1
   Compiling resolv-conf v0.7.0
   Compiling parking_lot v0.11.2
   Compiling uuid v0.8.2
   Compiling ordered-float v2.8.0
   Compiling flate2 v1.0.22
   Compiling rand_chacha v0.2.2
   Compiling digest v0.9.0
   Compiling block-buffer v0.9.0
   Compiling crypto-mac v0.10.1
   Compiling regex v1.5.4
   Compiling url v2.2.2
   Compiling crossbeam-deque v0.8.1
   Compiling chrono v0.4.19
   Compiling sha-1 v0.9.8
   Compiling sha2 v0.9.8
   Compiling rand v0.7.3
   Compiling hmac v0.10.1
   Compiling brotli2 v0.3.2
   Compiling webpki v0.21.4
   Compiling sct v0.6.1
   Compiling webpki-roots v0.20.0
   Compiling futures-macro v0.3.18
   Compiling thiserror-impl v1.0.30
   Compiling derive_more v0.99.17
   Compiling actix-macros v0.1.3
   Compiling time-macros-impl v0.1.2
   Compiling pin-project-internal v1.0.8
   Compiling enum-as-inner v0.3.3
   Compiling derivative v2.2.0
   Compiling time-macros v0.1.1
   Compiling thiserror v1.0.30
   Compiling pin-project v0.4.28
   Compiling pin-project v1.0.8
   Compiling actix-service v1.0.6
   Compiling futures v0.3.18
   Compiling serde_urlencoded v0.7.0
   Compiling serde-value v0.7.0
   Compiling serde_yaml v0.8.23
error[E0107]: this struct takes 3 generic arguments but 2 generic arguments were supplied
  --> /home/dfmb/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_yaml-0.8.23/src/mapping.rs:16:10
   |
16 |     map: IndexMap<Value, Value>,
   |          ^^^^^^^^ -----  ----- supplied 2 generic arguments
   |          |
   |          expected 3 generic arguments
   |
note: struct defined here, with 3 generic parameters: `K`, `V`, `S`
  --> /home/dfmb/.cargo/registry/src/github.com-1ecc6299db9ec823/indexmap-1.7.0/src/map.rs:76:12
   |
76 | pub struct IndexMap<K, V, S> {
   |            ^^^^^^^^ -  -  -
help: add missing generic argument
   |
16 |     map: IndexMap<Value, Value, S>,
   |                               +++

For more information about this error, try `rustc --explain E0107`.
error: could not compile `serde_yaml` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ keywords = ["yaml", "serde"]
 
 [dependencies]
 ryu = "1.0"
-indexmap = "1.5"
+indexmap = { version = "1.5.2", features = ["std"] }
 serde = "1.0.69"
 yaml-rust = "0.4.5"
 

